microsoft/testsuites/sysbench/sysbenchperf.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+from pathlib import Path
+from typing import Any, Dict, List
+
+from assertpy.assertpy import assert_that, fail
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+)
+from lisa.messages import TestStatus, send_sub_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import Sysbench
+from lisa.tools.sysbench import SysbenchTestResult
+
+
+@TestSuiteMetadata(
+    area="sysbench",
+    category="performance",
+    description="""
+    This test suite is for executing the sysbench tests
+    """,
+)
+class SysbenchTestSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Runs Sysbench test for cpu
+        """,
+        priority=3,
+    )
+    def perf_sysbench_cpu(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        subtest_name: str = "default_cpu_perf_check"
+        message = ""
+        status = TestStatus.QUEUED
+        output: SysbenchTestResult = node.tools[Sysbench].run_cpu_perf()
+
+        if output.exit_code != 0:
+            message = f"{subtest_name} failed with exit code {output.exit_code}"
+            status = TestStatus.FAILED
+        else:
+            message = json.dumps(output.result)
+            status = TestStatus.PASSED
+
+        self._send_subtest_msg(
+            result,
+            subtest_name,
+            status,
+            message,
+        )
+
+        msg = f"Sysbench-cpu test failed with exit code {output.exit_code}"
+        assert_that(status).described_as(msg).is_equal_to(TestStatus.PASSED)
```

Comment:
The whole part from sending subresults to assert the exit code can be moved into the tool. The test case level is for test definition and configurations.

Reply:
Sure, Will make this change

Reply:
I have made changes to push the sending subresults to assert the exit code to tool level. Please check